src/scope.ml

```diff
@@ -120,38 +120,93 @@ type mode =
       carries the environment for variables that will be bound in the
       representation of the RHS. *)
 
+(** [get_implicitness t] gives the specified implicitness of the parameters of
+    a symbol having the (parser-level) type [t]. *)
+let get_implicitness : p_term -> bool list = fun t ->
+  let rec get_impl t =
+    match t.elt with
+    | P_Prod(xs,t) -> List.map (fun (_,_,impl) -> impl) xs @ get_impl t
+    | P_Impl(_,t)  -> false :: get_impl t
+    | P_Wrap(t)    -> get_impl t
+    | _            -> []
+  in
+  let impl = get_impl t in
+  (* Minimization of the data: the list does not end with [false]. *)
```

Comment:
Explanation missing.

Reply:
What kind of explanation are you thinking about?

I maintained an invariant that is not documented: the `sym_implicits` field of symbols contains a list that is empty, or does not have `false` as its last element. We are only interested in having information about implicit arguments. The explicit (i.e., non-implicit) arguments that do not appear "in between" implicit arguments do not need to make it to the list.

Reply:
As you said: this is not documented ;-) Could you please add a comment in terms.ml expliciting the invariant and, here, that it is done for ensuring the invariant?